src/common/structures/apiTree.ts

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { Name } from "./name";
+import fs from "fs-extra";
+import path from "path";
+import { ramlToolLogger } from "../logger";
+
+/*
+ * Api Tree is an abstract class that represents the directory structure of several APIs
+ *
+ */
+export abstract class ApiTree {
+  metadata: { [key: string]: unknown };
+
+  constructor(
```

Comment:
I see that constructor is the only function that this class has. If it is the only common code I would suggest to move it to a helper function and avoid inheritance. As the code grows inheritance makes code complicated and hard to reuse

Reply:
I agree...and looking at some of wil's comments I think Im actually going to rename "ApiTree -> ApiMetadata" and remove this class and make it no longer abstract.  Then rename "Api -> ApiModel" which will extend ApiTree.   @wjhsf and @unandyala does this make more sense to both of you?

Reply:
Does ApiModel need to extend ApiMetadata? Or is it simpler to say ApiModel has/contains ApiMetadata

Reply:
What will you do with the existing ApiMetadata class?